src/main/java/org/opensearch/knn/index/VectorField.java

```diff
@@ -8,28 +8,21 @@
 import org.apache.lucene.document.Field;
 import org.apache.lucene.index.IndexableFieldType;
 import org.apache.lucene.util.BytesRef;
+import org.opensearch.knn.index.codec.VectorSerializer;
+import org.opensearch.knn.index.codec.VectorSerializerFactory;
 
-import java.io.ByteArrayOutputStream;
-import java.io.ObjectOutputStream;
+import static org.opensearch.knn.index.codec.VectorSerializerFactory.SerializationMode.COLLECTION_OF_FLOATS;
 
 public class VectorField extends Field {
 
     public VectorField(String name, float[] value, IndexableFieldType type) {
         super(name, new BytesRef(), type);
         try {
-            this.setBytesValue(floatToByte(value));
+            final VectorSerializer vectorSerializer = VectorSerializerFactory.getSerializerBySerializationMode(COLLECTION_OF_FLOATS);
```

Comment:
Wondering if it might be better to use a default here and set the default in VectorSerializerFactory. Whatre your thoughts @martin-gaievski ?

Reply:
Just to make sure my understanding is correct - are you suggesting to remove parameter from getSerializerBySerializationMode, so factory is alway return VectorAsCollectionOfFloatsSerializer?

Reply:
Discussed offline, intention is to have explicit method to get default implementation

Reply:
made the change, resolving